test/integration/model.test.js

```diff
@@ -600,6 +600,23 @@ describe(Support.getTestDialectTeaser('Model'), () => {
         expect(idx2.unique).not.to.be.ok;
       }
     });
+
+    it('should allow custom timestamps with underscored: true', async function () {
+      const User = this.sequelize.define('UserCustomTimestampUnderscored', {}, {
+        createdAt: 'createdOn',
+        updatedAt: 'modifiedOn',
+        underscored: true,
+      });
+
+      expect(User.rawAttributes).to.haveOwnProperty('createdOn');
+      expect(User.rawAttributes).to.haveOwnProperty('modifiedOn');
```

Comment:
Maybe the issue is the DB name:

```suggestion
      expect(User.rawAttributes).to.haveOwnProperty('modifiedOn');
      expect(User.rawAttributes.modifiedOn.field).to.eq('modified_on');
```

Reply:
What do you mean with the DB name? Or is that unrelated to your suggestion?

Reply:
The key in `rawAttributes` is the JS name of the property. Since you mentioned this test passes in sqlite, it's possible that the issue is that the column name (db side) is incorrectly generated (as this is an issue with `underscored` apparently). `User.rawAttributes.modifiedOn.field` returns which name will be used for the column

Reply:
sqlite was just the first dialect I tried but the test passed for other dialects as well. I'm not sure what the failing test is that @evanrittenhouse was able to come up with.

Even with the added `User.rawAttributes.modifiedOn.field` it seems to pass the test. So I'm not sure what's wrong here

Reply:
@WikiRik I saw it fail when I put this test https://github.com/sequelize/sequelize/blob/af0f024acdcb4f885b9969a958c5fb4752abfff8/test/unit/model/define.test.js#L15-L31 into an integration test. Unfortunately, I don't have more information as it was uncommitted work on the branch that I've been using to work on the new issue associated with #14723.

Reply:
I think I used that test, possibly without `timestamp: true`, as well and it passed on CI. But I can try it again tomorrow

Either way, I think it might also be related to the usage of sequelize-typescript as I've asked in the original issue

Reply:
Nope, that passes as well so I'll close this PR since I can't seem to reproduce this error